cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java

```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
                     + "-  OK  - 'collection_prefix=" + collectionPrefix
-                    + "' correctly encoded as '" + encodedCollectionPrefix + "'");
+                    + "' correctly encoded as '" + expectedCollectionPrefix + "'");
         } catch (AssertionError e) {
             System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
                     + "- FAIL - 'collection_prefix=" + collectionPrefix
-                    + "' wrongly encoded as '" + encodedCollectionPrefix + "'");
+                    + "' wrongly encoded as '" + expectedCollectionPrefix + "'");
             throw e;
-        } // try catch
-    } // testConfigureCollectionPrefixIsEncoded
+        } // try catch // try catch
+    } // testConfigureCollectionPrefixIsEncodedOldEncoding
+    
+    /**
+     * [NGSIMongoSink.configure] -------- Configured 'collection_prefix' is encoded when having forbiden characters.
+     */
+    @Test
+    public void testConfigureCollectionPrefixIsEncodedNewEncoding() {
+        System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
+                + "-------- Configured 'collection_prefix' is encoded when having forbiden characters");
+        String collectionPrefix = "this\\is/a$prefix.with-forbiden,chars:-.";
+        String dbPrefix = "sth_";
+        String dataModel = null; // default
+        String enableEncoding = "true";
+        NGSIMongoBaseSinkImpl sink = new NGSIMongoBaseSinkImpl();
+        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel,
+                enableEncoding));
+        String expectedCollectionPrefix = NGSICharsets.encodeMongoDBCollection(collectionPrefix);
+        
+        try {
+            assertEquals(expectedCollectionPrefix, sink.collectionPrefix);
+            System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
+                    + "-  OK  - 'collection_prefix=" + collectionPrefix
+                    + "' correctly encoded as '" + expectedCollectionPrefix + "'");
+        } catch (AssertionError e) {
+            System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
+                    + "- FAIL - 'collection_prefix=" + collectionPrefix
+                    + "' wrongly encoded as '" + expectedCollectionPrefix + "'");
+            throw e;
+        } // try catch // try catch
+    } // testConfigureCollectionPrefixIsEncodedNewEncoding
     
     /**
      * [NGSIMongoSink.configure] -------- Configured 'db_prefix' is encoded when having forbiden characters.
      */
     @Test
-    public void testConfigureDBPrefixIsEncoded() {
+    public void testConfigureDBPrefixIsEncodedOldEncoding() {
         System.out.println(getTestTraceHead("[OrionMongoBaseSink.configure]")
                 + "-------- Configured 'db_prefix' is encoded when having forbiden characters");
         String collectionPrefix = "sth_";
         String dbPrefix = "this\\is/a$prefix.with forbiden\"chars:-,";
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4